TaskServiceInterview.cs

```diff
@@ -0,0 +1,448 @@
+
+            if (!hasUserClassification)
+            {
+                taskForward.AssigneeInitials = null;
+            }
+
+            if (hasUserClassification)
+            {
+                taskForward.AssigneeInitials = employeeInitials;
+            }
+
+            if (string.IsNullOrEmpty(taskForward.AssigneeInitials))
+            {
+                task.AssigneeInitials = null;
+                task.Status = TaskStatus.New;
+            }
+            else
+            {
+                task.AssigneeInitials = taskForward.AssigneeInitials;
+            }
+
+            task.SetClassificationAndRecalculatePermission(this.classificationRepository.Get(taskForward.ClassificationId));
+            task.IsHighPriority = taskForward.IsHighPriority;
+
+            // clear data requested by user
+            task.LockedByInitials = null;
+            task.LockedByTimestamp = null;
+
+            this.taskValidator.Validate(task, TaskValidationOperations.Forward);
+
+            var lockEventTime = this.GetLockTime(task.Id, employeeInitials, now);
+
+            var forwardBusinessEventData = new ForwardBusinessEventData
+            {
+                NewClassificationId = taskForward.ClassificationId.ToString(),
+                OldClassificationId = oldClassification.Id.ToString(),
+                NewOwner = taskForward.AssigneeInitials,
+                OldOwner = old,
+                TimeToCompleteTicks = (now - lockEventTime).Ticks,
+                DoneAfterDeadline = task.Deadline < now
+            };
+
+            task.SetActionExecutorInformation(ChangeReason.Forward, employeeInitials, this.clock.Current);
+            task.AddBusinessEventForward(employeeInitials, now, forwardBusinessEventData);
+
+            this.auditLogServant.Log("Task forwarded", new { taskForward, employeeInitials });
+            this.taskChangeNotificationServant.NotifyClassificationChanged(task, oldClassification);
+
+            this.notificationService.NotifyTaskOwnerAboutTaskAssignment(task, employeeInitials);
+
+            return true;
+        }
+
+        private TaskDetailsItem MapToTaskDetails(TaskEntity task)
+        {
+            var taskDetails = this.mappingServant.Map<TaskEntity, TaskDetailsItem>(task);
+            taskDetails.LockedByInitials = this.taskLockServant.IsLocked(task) ? task.LockedByInitials : null;
+            return taskDetails;
+        }
+
+
+        private IEnumerable<TaskListItem> ConvertToTaskListItems(IEnumerable<TaskEntity> tasks)
+        {
+            foreach (var task in tasks)
+            {
+                var taskListItem = this.mappingServant.Map<TaskEntity, TaskListItem>(task);
+                taskListItem.InternalReminderCounter = this.internalReminderFinder.GetNumberOfTaskInternalReminders(task.Id);
+                taskListItem.LockedByInitials = this.taskLockServant.IsLocked(task) ? task.LockedByInitials : null;
+                yield return taskListItem;
```

Comment:
imo unikalbym yielda, bo wiele osob moze go nie znac i przez to sprawia, ze kod jest mniej czytelny